src/services/leafletWatchHelpers.js

```diff
@@ -20,7 +20,17 @@ angular.module("leaflet-directive")
   to set watching to once and its watch depth.
   */
   var _maybeWatch = function(scope, thingToWatchStr, watchOptions, initCb){
-      return _maybe(scope, '$watch', thingToWatchStr, watchOptions, initCb);
+      var localWatchOptions, watchMethod;
```

Comment:
Yes, I see your point. `doWatch`, at least as the name suggests, really should only accept a boolean. However, I think the reason I went with this approach (I actually made this change a while ago) was that the watching of an object isn't necessarily a binary parameter and I interpreted `doWatch` as the appropriate interface for modifying the watch behaviour.

How about instead of `isDeep` and `doWatch`, we add a property `type` which can be either: `deep`, `collection`, `normal`, `no`. 

`deep` = $watch(..., true)
`collection` = $watchCollection(...)
`normal` = $watch(...)
`no` = no watch set

I'm not set on the names `normal` and `no`, so if you have better ideas let me know. I think the advantage here is that we open ourselves up to even more hypothetical watching strategies in the future (e.g. watch by a marker property instead of the marker itself). Of course, we'd have to make this backwards compatible with doWatch and isDeep (if doWatch and isDeep exist they could override type... or vice versa).

Thoughts?